patches/com/mojang/blaze3d/systems/RenderSystem.java.patch

```diff
@@ -0,0 +1,23 @@
+--- a/com/mojang/blaze3d/systems/RenderSystem.java
++++ b/com/mojang/blaze3d/systems/RenderSystem.java
+@@ -1025,4 +1025,20 @@
+          void accept(it.unimi.dsi.fastutil.ints.IntConsumer p_157488_, int p_157489_);
+       }
+    }
++
++   public static void backupGlState(net.neoforged.neoforge.client.GlStateBackup state) {
++      if (!isOnRenderThread()) {
```

Comment:
I think this should really be `assertOnRenderThread()` instead, otherwise the `state` will not be written correctly. And same for `restoreGlState`.

Reply:
Yeah, that sounds reasonable in this case considering this would only be used in immediate-mode stuff like UIs anyway.